tests/installation/await_install.pm

```diff
@@ -81,44 +81,47 @@ sub wait_countdown_stop {
     return wait_screen_change(undef, $stilltime, similarity => $similarity);
 }
 
+sub _set_timeout {
+    my ($timeout) = $_[0];
+    # upgrades are slower
+    ${$timeout} = 5500 if (get_var('UPGRADE') || get_var('LIVE_UPGRADE'));
+    # our Hyper-V server is just too slow
+    # SCC might mean we install everything from the slow internet
+    ${$timeout} = 5500 if (check_var('VIRSH_VMM_FAMILY', 'hyperv') || (check_var('SCC_REGISTER', 'installation') && !get_var('SCC_URL')));
+    # VMware server is also a bit slow, needs to take more time
+    ${$timeout} = 3600 if (check_var('VIRSH_VMM_FAMILY', 'vmware'));
+
+    # aarch64 can be particularily slow depending on the hardware
+    ${$timeout} *= 2 if check_var('ARCH', 'aarch64') && get_var('MAX_JOB_TIME');
+    # PPC HMC (Power9) performs very slow in general
+    ${$timeout} *= 2 if check_var('BACKEND', 'pvm_hmc') && get_var('MAX_JOB_TIME');
+    # encryption, LVM and RAID makes it even slower
+    ${$timeout} *= 2 if (get_var('ENCRYPT') || get_var('LVM') || get_var('RAID'));
+    # "allpatterns" tests install a lot of packages
+    ${$timeout} *= 2 if check_var_array('PATTERNS', 'all');
+    # multipath installations seem to take longer (failed some time)
+    ${$timeout} *= 2 if check_var('MULTIPATH', 1);
+    # Scale timeout
+    ${$timeout} *= get_var('TIMEOUT_SCALE', 1);
+}
+
 sub run {
     my $self = shift;
     # NET isos are slow to install
     my $timeout = 2000;
 
     # workaround for yast popups and
     # detect "Wrong Digest" error to end test earlier
-    my @tags = qw(rebootnow yast2_wrong_digest yast2_package_retry yast_error initializing-target-directory-failed);
+    my @tags = qw(rebootnow yast2_wrong_digest yast2_package_retry yast_error initializing-target-directory-failed linuxrc_error);
+    _set_timeout(\$timeout);
     if (get_var('UPGRADE') || get_var('LIVE_UPGRADE')) {
         push(@tags, 'ERROR-removing-package');
         push(@tags, 'DIALOG-packages-notifications');
         # There is a dialog with packages that updates are available from
         # the official repo, do not use those as want to use not published repos only
         push(@tags, 'package-update-found') if is_opensuse;
-        # upgrades are slower
-        $timeout = 5500;
+        # upgrades are slower so we $timeout = 5500;
```

Comment:
something seems to have gone wrong here

Reply:
The timeout moved into the `_timeout` which handles all the timeouts. i left the comment as the reference. But the comment might be no clear enough. i will rephrase it